### ISSUE TITLE
[Feature] Azure AD Workload Identity Managed AKS Addon

### ISSUE THREAD

**Issue by tomkerkhove** [Azure/AKS#1480]
It would be good to have built-in support for [AAD Pod Identity](https://github.com/Azure/aad-pod-identity) for which we can opt-out during creation of AKS cluster.

By doing that, we could :
- Assign Azure AD identities to AKS instance and automatically register them inside the cluster
- Browse the Azure portal to see which Azure AD identities are available inside the cluster

**Comment by sanjeebsarangi** [comment]
Currently, we use AAD Pod Identity and would love to use this feature.

**Comment by JohnGalt1717** [comment]
Please make sure that this is all part of the portal UI as well. As an example the integration with Azure Container Registries exists, but you have to use AZ CLI instead of just clicking on a drop down and adding it.

**Comment by tomkerkhove** [comment]
Definately agree about that!

**Comment by jluk** [comment]
We hear you, portal integration is constantly improving and another wave is in the works. 
cc @jenetlan

**Comment by syedhassaanahmed** [comment]
With the GA of Managed Identity for AKS Clusters, it has become even more [painful](https://github.com/Azure/aad-pod-identity/blob/master/docs/readmes/README.msi.md#pre-requisites---role-assignments) to install AAD Pod Identity. Would be great to have this feature released soon.

**Comment by jenetlan** [comment]
@JohnGalt1717 and other - thanks for the feedback :) Just wanted to quickly chime in and let you know that we're actively working on integration between AKS and ACR in portal. We're focusing on setting up that connection when deploying a new AKS cluster to start with - is that what you had in mind or were you thinking more about connecting an ACR to an existing AKS cluster?

**Comment by JohnGalt1717** [comment]
@jenetlan I'd like to see both.

Thinking like an Add-ons page after creation that lists all of the addons like ACR (add one or more), AAD-POD identity, Azure Application Gateway, Public IP, Let's Encrypt with integration with Azure DNS, Nginx Ingress, Key Vault, etc linked to Public IP, all in the Addons menu as a start of add-ons that can be configured both with a single line with az aks and in portal on it's on add-ons tab that would allow you to add and pick from a list of available, fill out whatever you need to fill out and you're good to go.

This stuff is incredibly accident prone currently with long list of steps that could be cut down if these were easy to install from CLI and portal and they just setup everything for you automatically. 

Better, let 3rd parties also create addons that let them offer other stuff too that directly works with Azure in k8s.

**Comment by tomkerkhove** [comment]
Fully agreed, thinks like KEDA (keda.sh) would be nice as an add-on as well!

However, the biggest pain for me is still AAD Pod Identity.

**Comment by gd-asharov** [comment]
@jluk do you have any news about this issue?

**Comment by se7enack** [comment]
@jluk Anything on this?

**Comment by PixelRobots** [comment]
I see this has moved to public preview. Do we have any docs on this add-on yet?

**Comment by PixelRobots** [comment]
Think I found it. https://docs.microsoft.com/en-us/azure/aks/use-azure-ad-pod-identity

**Comment by tomkerkhove** [comment]
This is super helpful and makes it a lot easier, thanks team!

**Comment by miwithro** [comment]
Welcome.  I will close the issue now.

**Comment by benjamincburns** [comment]
Is this preview feature available in all regions where AKS is available, or is it a more limited set? It would be nice if preview feature availability by region was documented somewhere. Apologies if it is - I went looking but couldn't find it.

**Comment by miwithro** [comment]
Yes it is available in all regions.

**Comment by jrauschenbusch** [comment]
Is there an ETA when this feature will goes GA?

**Comment by miwithro** [comment]
The current implementation of Pod Identity (V1) is currently in Public Preview with CSS support.   We are working on Pod Identity (V2) which will leverage the native K8s OIDC federation.  Pod Identity V2 will go Public Preview in June 2021.  The intention is to take that to GA by end of the calendar year, and then deprecate Pod Identity V1.

**Comment by PiotrZalewski-TomTom** [comment]
What about GA for V1?  Is this in plans or need to wait for V2?

**Comment by miwithro** [comment]
@PiotrZalewski-TomTom we have no plans to GA V1.  We onboarded V1 to support while we work to GA V2.

**Comment by dvob** [comment]
@miwithro 
> ... Pod Identity (V2) which will leverage the native K8s OIDC federation.

Does this mean with V2 we can use service account tokens (projected tokens) to access azure resources? Or is there somewhere a document which describes how V2 is going to work? Since we are using V1 extensively we would like to know what we have to expect with V2.

**Comment by miwithro** [comment]
@dvob we are still working on the design, but yes that is the desire.

We are building "service identity federation" which will allow developers to provide AAD with a token from a foreign trusted issuer.

**Comment by ekhaydarov** [comment]
If people use V1 will you just completely remove it like you removed the pod security policy preview for new clusters breaking workflows?

**Comment by miwithro** [comment]
@ekhaydarov the plan is to run V1 and V2 in parallel, and seamlessly migrate the Identities from V1 to V2.

**Comment by whiskeysierra** [comment]
But it sounded like in V2 there wouldn't be managed identities anymore but that kubernetes service accounts could authenticate as themselves directly to interact with Azure?

**Comment by benc-uk** [comment]
I don't think that's possible, if the identity doesn't exist in Azure & AAD there is no way it can't interact with Azure or the APIs (e.g. ARM)

**Comment by alexl2012** [comment]
Hello, please can i bring your attention to 

#2221

**Comment by mosabami** [comment]
Is the ability to enable AAD Pod Identity on already existing cluster in development?

**Comment by miwithro** [comment]
Yes you can do that today.

https://docs.microsoft.com/en-us/azure/aks/use-azure-ad-pod-identity#update-an-existing-aks-cluster-with-azure-cni

**Comment by kvendingoldo** [comment]
any news?

**Comment by sergiibielskyi** [comment]
Hi, we are working on a specific project with AKS (windows nodes) to host a legacy system for the client. But we faced that one feature in preview that we would like to use named managed identities in Azure Kubernetes Service. The question here is that any roadmap or deadline where this particular feature will be released? Big thanks in advice.

**Comment by miwithro** [comment]
V2 which works on Windows goes into Public Preview on OSS in January.  We hope to be in Public Preview in AKS by the end of Q2 with a tentative GA date of Q3.

**Comment by sergiibielskyi** [comment]
> V2 which works on Windows goes into Public Preview on OSS in January. We hope to be in Public Preview in AKS by the end of Q2 with a tentative GA date of Q3.

Thank you for the quick answer. That means it could be dangerous to use this feature in production right now. Or can we?

**Comment by miwithro** [comment]
For the OSS version you can get support from the Upstream team.  Once it is Public Preview you will have Microsoft support.

**Comment by kvendingoldo** [comment]
will v2 version work w/o CRD based on K8S annotations or not?

**Comment by aramase** [comment]
> will v2 version work w/o CRD based on K8S annotations or not?

Hello 👋🏻  

AAD Pod Identity v2 will not require any CRD for identity configuration. The configuration will be done by annotating the Kubernetes service account.

**Comment by PixelRobots** [comment]
https://azure.github.io/azure-workload-identity/docs/

**Comment by darthmolen** [comment]
@miwithro So why did this consume pod identity v2? I'm confused here, what moved this back to "in-development"?

**Comment by PixelRobots** [comment]
The issue was called pod identity. And was in public preview. This issue got renamed to its current name and moved back to in progress.

The new solution is not called pod Id v2 so the rename of the issue was to try and stop confusion about the pod ID v2 naming I believe.

**Comment by miwithro** [comment]
@darthmolen @PixelRobots is correct. We have rebranded Pod Identity V2 to Azure AD Workload Identity.

**Comment by kvendingoldo** [comment]
@miwithro do you have any estimates when azure pod identity v1 can be replaced with azure workload identity for production environments?

**Comment by PixelRobots** [comment]
Looking at quarter 3. But no date is set in stone. Feedback can change things.

**Comment by dragan-novakovic** [comment]
When can we expect public preview ?

**Comment by miwithro** [comment]
We are shooting for April for Public Preview on AKS.

**Comment by ngbrown** [comment]
How will this enable getting secret contents stored in an Azure Key Vault and mount them into Kubernetes pods as files or Kubernetes secrets? Azure Key Vault Provider for Secrets Store CSI Driver was the way to do it with v1.

The v2, or Azure AD Workload Identity, documentation makes no mention of connecting the Key Vault secrets to pods as files or native Kubernetes Secrets.

Is this functionality in scope for v2?

**Comment by Krupakarreddin** [comment]
@PixelRobots How to create the federated credential in a service principal through cli?..if we want to automate

**Comment by miwithro** [comment]
@Krupakarreddin https://azure.github.io/azure-workload-identity/docs/topics/federated-identity-credential.html

**Comment by miwithro** [comment]
@ngbrown once we GA Workload Identity we will update the guidance for integrating with CSI Secret Store

**Comment by Krupakarreddin** [comment]
@miwithro is azwi supported in china region .I was getting 404 error 
 No subscription provided, using selected subscription from Azure CLI: xxxxxxxxxxxxxxxxx
Error: Unexpected response from Get Subscription: 404

**Comment by miwithro** [comment]
@Krupakarreddin not yet.

**Comment by sozercan** [comment]
@Krupakarreddin can you try with `--azure-env=AzureChinaCloud` flag?

**Comment by Krupakarreddin** [comment]
@sozercan In the Azure China Cloud, federated credentials option itself is not seen in the Azure AD ServicePrincipal

**Comment by kwinyyyc** [comment]
@miwithro May I know is it in public preview now?

**Comment by miwithro** [comment]
Not yet, we are working on it. I will post an update to this issue when it moves to Public Preview.  So you are watching the right place.

**Comment by SychevIgor** [comment]
2 years, still in progress.

**Comment by miwithro** [comment]
We are targeting 9/30 for Public Preview.  That is when the Federated Identity support in Managed Identity will go to Public Preview.

**Comment by MaroWielki** [comment]
Hi,
Will the use case of "Federated Identity support in Managed Identity" be similar to the one described in aad-pod-identity ?

https://docs.microsoft.com/en-au/azure/aks/use-azure-ad-pod-identity

**Comment by miwithro** [comment]
@MaroWielki Federated Identity support in Managed Identity is a dependency for Azure AD Workload Identity.  Federated Identity was already added to App Registration, now we are working on adding it to Managed Identity.  Once it is Public Preview/GA on the Identity side, we will move to Public Preview/GA on the AKS side.

**Comment by kvendingoldo** [comment]
@miwithro will it be v1 or v2 version on public preview?

**Comment by miwithro** [comment]
@kvendingoldo V1 is pod identity which is currently in Public Preview.  V2 is Workload Identity which is currently in Private Preview, and will go to Public Preview than GA.  Then we will work to deprecate V1.

**Comment by MaroWielki** [comment]
> @MaroWielki Federated Identity support in Managed Identity is a dependency for Azure AD Workload Identity. Federated Identity was already added to App Registration, now we are working on adding it to Managed Identity. Once it is Public Preview/GA on the Identity side, we will move to Public Preview/GA on the AKS side.


Does this mean that applications in pods will access other Azure services via app registrations with MI account privileges?

**Comment by miwithro** [comment]
No it is either use App Registration w/Federated Identity or Managed Identity w/Federated Identity.

**Comment by sherdana** [comment]
> We are targeting 9/30 for Public Preview. That is when the Federated Identity support in Managed Identity will go to Public Preview.

Where can I see private preview documentation?

**Comment by miwithro** [comment]
@sherdana do you have an email I can send it to you?

**Comment by manali1st** [comment]
> 

Hello @miwithro , as you stated earlier is it still on track for GA - 9/30 - for Workload identity release date

**Comment by miwithro** [comment]
@manali1st yes it is.

**Comment by PixelRobots** [comment]
> @manali1st yes it is.

GA with managed identity support?

**Comment by miwithro** [comment]
We are targeting 12/30.

**Comment by asanjabi** [comment]
So public preview 9/30 and GA 12/30? 

manali1st's question asked about GA.

**Comment by dciborow** [comment]
@miwithro , interested in more details about your Public Preview for my own internal usage. Could you ping me to chat? dciborow@microsoft.com

**Comment by mifurm** [comment]
@tomkerkhove - does this new feature will also favour Azure CNI over kubnet with respect to security? as with Managed Identity for AKS? (link for references: https://docs.microsoft.com/en-us/azure/aks/use-azure-ad-pod-identity#using-kubenet-network-plugin-with-azure-active-directory-pod-managed-identities)

**Comment by tomkerkhove** [comment]
No idea, I'm not part of this team; sorry.

**Comment by pinkfloydx33** [comment]
I would think it shouldn't matter. With pod identity, the network requests are intercepted by mic/nmi and translated into a request out to azure for a token. It has to know who/what you are based on labels on your pod as it works with an intercepted/rerouted request. (very simplistic explanation). ARP spoofing could cause podIdentity to think you're a total different pod. 

With AZWI you get federated credentials based on the Kubernetes service account you're running under, and not some random aspect of the pod making the request. The requests aren't intercepted and rerouted. It's much more secure in general.

**Comment by miwithro** [comment]
@mifurm Pod Identity is IMDS based on NMI/MIC use the node identity to translate the identity on the pods behalf.  Workload Identity is OIDC based, and with Service Account Token Volume Projection now you can inject a token of an identity directly into the pod, so there is no need for a node identity anymore.

**Comment by MaroWielki** [comment]
> @sherdana do you have an email I can send it to you?

@miwithro can you share the docs with me (marek649spam@gmail.com) please?

**Comment by miwithro** [comment]
@MaroWielki we are going Public Preview on 9/30 we will have docs on AKS docs.

**Comment by asubmani** [comment]
Is there documentation on how to migrate to workload identity. It seems there will be 2 migrations
1) For replacing the existing OSS implementation with the addon (by the AKS operator); and remove the CRDs
2) Some change might be needed in the application code. Are there any examples of `old vs new` for popular languages (Java, C# etc.?)


All of the apps are Java or Spring based apps and use MSICredential to access Key Vault, and other resources. we have ~650 MSI's being managed by the OSS version.

** Will there be any limits to the number of workload identities we can create per subscription/AKS cluster? **

**Comment by miwithro** [comment]
@asubmani yes there will be migration documents.  You are correct there are a couple of scenarios that will be covered.

1. Greenfield deployment
a. Using nothing before
b. Using an outdated sdk
2. Brownfield deployment
a. OSS migration
b. Pod Identity migration
c. using an outdated sdk

**Comment by KnicKnic** [comment]
> @MaroWielki we are going Public Preview on 9/30 we will have docs on AKS docs.

link?

**Comment by timja** [comment]
It's https://learn.microsoft.com/en-us/azure/aks/workload-identity-overview#how-to-migrate-to-workload-identity

**Comment by asubmani** [comment]
> https://learn.microsoft.com/en-us/azure/aks/workload-identity-overview#how-to-migrate-to-workload-identity

What about sequence of uninstalling the existing aadpodidentity (we are not talking about what developers need to do; but what the Kubernetes operator who will remove the existing aadpodidentity and next enable the workload identity need to do) I fail to see how existing pods that are looking for aadpodidentity references will continue to work?

This is where 90% of AKS customers are who use podidentity today..  How do they migrate?

Wouldn't we need to un-install the old manually installed `aadpodidenty` first?  If yes, what is the migration strategy? Scenario 1 doesn't really work without a migration guide on the "infra side"

We have close to 630 User MSI's created in Azure and a lot of pods using them. We manually assign these identities in the AKS VMSS.  I presume that when you remove the AAD Podidentity, all authentication will temporarily fail till workload identity get's enabled.

**Comment by pinkfloydx33** [comment]
As someone doing this currently, you can leave the podIdentity labels and get the the service account setup etc. This should be enough since the apps will now start using the oidc/federated token and not make (intercepted) requests out to imds/mic/nmi. You can follow it up with a pod identity exception for good measure and finally removing podid labels and deletion of the pod-identity (if using aks managed pod-identity v1). I had zero downtime while migrating my existing apps during private preview.

**Comment by asubmani** [comment]
> As someone doing this currently, you can leave the podIdentity labels and get the the service account setup etc. This should be enough since the apps will now start using the oidc/federated token and not make (intercepted) requests out to imds/mic/nmi. You can follow it up with a pod identity exception for good measure and then finally remove the podid labels and deletion of the pod-identity (if using aks managed pod-identity v1). I had zero downtime while migrating my existing apps during private preview.

Do you have a blog post or some high level overview of the steps? Did you have to make any changes to kubernetes deployments that reference the aadpodidently label?

**Comment by pinkfloydx33** [comment]
No blog, sorry... But it's rather simple:

1. Create federated credentials in Azure
2. Add labels and annotations to Service Account associated with deployment
3. Restart deployment

You can remove the pod identity labels whenever (or never). The above is dependent on your app using newer versions of the azure identity / msal libraries. For example if you have a dotnet app connecting to SQL server you need  Azure.Identy 1.5.0-beta4+ (or just Ms.Data.SqlClient 5.0+). The underlying libs must support it. In my case everything was already up to date so it was smooth sailing.

**Comment by asanjabi** [comment]
@asubmani This might be helpful for you: https://learn.microsoft.com/en-us/azure/aks/workload-identity-migration-sidecar

**Comment by asubmani** [comment]
> * ure
> * Add labels and annotations to Service Account associated with deployment
> * Restart deployment

Thanks @pinkfloydx33; would the number of identities be a concern here? I am asking because in the previous Public Preview last year there were serious limitations; (50 Managed Identities per cluster etc.)

**We have over 630 Identities in a single cluster. Is there any limit to the number of identities "Azure AD workload Identity Managed AKS Addon" will support?**
+ @aramase

**Comment by pinkfloydx33** [comment]
I don't think there's a limit to the number of identities as AZWI doesn't actually track those. There was (during private preview at least) a limit to the number of federated credentials per managed identity -- in rough terms, the number of k8s service accounts per identity

**Comment by miwithro** [comment]
@asubmani there is no limit on the number of Managed Identities but there is a limit (20) on the number of Federated Identities you can add to a Managed Identity.

**Comment by kishorerv25** [comment]
Any time line on this feature getting released in private cloud like AzureChinaCloud ??

**Comment by kishorerv25** [comment]
> Any time line on this feature getting released in private cloud like AzureChinaCloud ??

i just tested this feature in AzureChinaCloud.. and its working

**Comment by weisdd** [comment]
@AlexGoris-KasparSolutions not sure about the web-interface, but it's definitely available through CLI. You might need to update `aks-preview` first ([docs](https://learn.microsoft.com/en-us/azure/aks/workload-identity-deploy-cluster#install-the-aks-preview-azure-cli-extension)).

**Comment by AlexGoris-KasparSolutions** [comment]
> @AlexGoris-KasparSolutions not sure about the web-interface, but it's definitely available through CLI. You might need to update `aks-preview` first ([docs](https://learn.microsoft.com/en-us/azure/aks/workload-identity-deploy-cluster#install-the-aks-preview-azure-cli-extension)).

Thanks, that link helped me!

**Comment by dhirschfeld** [comment]
I'd like to use this capability on our production clusters - is there any timeline for when this feature will go GA?

**Comment by miqm** [comment]
> I'd like to use this capability on our production clusters - is there any timeline for when this feature will go GA?

https://github.com/Azure/AKS/issues/1480#issuecomment-1209778772

**Comment by miwithro** [comment]
GA is tracking for March 2023

**Comment by yorinasub17** [comment]
Hi, is there any official documentation or guidance on the OIDC Issuer URL format? I couldn't find any mention of what to expect for the issuer URL in the existing docs.

Context is that I am trying to use the [keyless signing feature of cosign](https://github.com/sigstore/cosign/blob/main/KEYLESS.md) which uses [fulcio](https://github.com/sigstore/fulcio). It seems to have been working earlier last year according to some PRs, but it looks like there's been a change to the OIDC Issuer URL format as what I have on my AKS cluster is different from what `fulcio` expects ([expected format](https://github.com/sigstore/fulcio/blob/f293398/config/fulcio-config.yaml#L52), while what I have is `https://eastus.oic.aks-prod.azure.com/***`). Would be great to have some documentation on the matter so that we aren't shooting in the dark with regards the format to expect on the tokens.

Many thanks!

**Comment by heoelri** [comment]
@miwithro the updated docs page is a bit unclear. `azure.workload.identity/use` is now mentioned in service account labels, pod labels as well as in pod annotations. Just guessing but I assume it's either (pod) labels OR annotations. The description is also a bit suboptimal: "Represents the service account
is to be used for workload identity." Expecting a boolean value of true/false. Would you mind verifying that?

**Comment by santiago-gilz** [comment]
> 

@miwithro will it be GA by March only in Azure Commercial or also in Azure Government?

**Comment by miwithro** [comment]
@heoelri azure.workload.identity/use true is now required in both the service account AND the pod.  

@santiago-gilz  GA in march in both Azure Commercial and Azure Government.

**Comment by heoelri** [comment]
> @heoelri Heyko Oelrichs FTE azure.workload.identity/use true is now required in both the service account AND the pod.

Thanks for the confirmation @miwithro! But is it on the pod-side [label](https://learn.microsoft.com/en-us/azure/aks/workload-identity-overview#pod-labels) **AND** [annotation](https://learn.microsoft.com/en-us/azure/aks/workload-identity-overview#pod-annotations) as the [docs article](https://learn.microsoft.com/en-us/azure/aks/workload-identity-overview) implies?

**Comment by miwithro** [comment]
@heoelri yes that is correct. Pod-side and annotation exactly as the doc suggests.

**Comment by tequeter** [comment]
@miwithro what about pods that only use a SecretProviderClass? The [CSI documentation](https://learn.microsoft.com/en-us/azure/aks/csi-secrets-store-identity-access#configure-workload-identity) has labels and annotations only on the Service Account, is that still valid?

**Comment by santiago-gilz** [comment]
@miwithro Are there any limitations at subscription or a higher scope? For instance a limit on the amount of Managed Identities or federated credentials (apart from the 20 per MI) you can create.

**Comment by ohorvath** [comment]
@miwithro  For our use case, the federated credentials / MI limit is a huge issue. In our clusters, we have 20-30 namespaces at least with 3-5 service account per cluster. We have to use a single MI for the AKS cluster as the MI is provided by our customers and we can't ask them to pre-create tons of MIs before cluster creation. We also can't create the MIs dynamically due to security concerns. Is there a plan to increase this limit? We need at least 100 federated credential per MI. With the current limitations we have to continue using the old AAD pod identity solution which is deprecated now.

**Comment by tom-scott** [comment]
> GA is tracking for March 2023

Is GA still on track for March?

**Comment by miwithro** [comment]
@tom-scott yes.

**Comment by tom-scott** [comment]
> @tom-scott yes.

Fantastic - thanks! 👍

**Comment by dpaardenkooper** [comment]
That sounds amazing! Looking forward to work with this!

**Comment by mblaschke-daimlertruck** [comment]
workload identity GA announcement was published yesterday but the announcement is not published anymore on [Azure updates](https://azure.microsoft.com/en-us/updates/) page, what happened?
also the link https://azure.microsoft.com/en-us/updates/ga-azure-active-directory-workload-identity-with-aks/ is not working anymore.

**Comment by PixelRobots** [comment]
> workload identity GA announcement was published yesterday but the announcement is not published anymore on [Azure updates](https://azure.microsoft.com/en-us/updates/) page, what happened?
> also the link https://azure.microsoft.com/en-us/updates/ga-azure-active-directory-workload-identity-with-aks/ is not working anymore.

Hello,

I posted this on twitter yesterday and @miwithro asked me to pull the tweet as it is now postponed till 14th April.

**Comment by Merlijnv** [comment]
Any update on GA? It was announced GA but then the announcement was deleted. But It was supposed to be GA in March but it's now April 😥

**Comment by Merlijnv** [comment]
Any succes? @PixelRobots

**Comment by PixelRobots** [comment]
> Any success? @PixelRobots 

I have just heard that it is now live and GA.

**Comment by mdaslamansari** [comment]
Where it is notified?

**Comment by PixelRobots** [comment]
> Where it is notified?

![PXL_20230418_083128006.MP.jpg](https://user-images.githubusercontent.com/22979170/232720365-8eee472e-a494-4ba9-b358-a44ff48cfdc7.jpg)

Just now at KubeCon. Docs are rolling out today hopefully.

**Comment by PixelRobots** [comment]
https://techcommunity.microsoft.com/t5/apps-on-azure-blog/general-availability-for-azure-active-directory-ad-workload/ba-p/3798292#M1196

**Comment by weisdd** [comment]
> https://techcommunity.microsoft.com/t5/apps-on-azure-blog/general-availability-for-azure-active-directory-ad-workload/ba-p/3798292#M1196

Thanks for sharing the link! My only concern is this:

> "Please note that during the Public Preview we enabled the sidecar to give customers time to update their SDKs, but **this is not a supported configuration for General Availability**."
Does it mean that pods will not be mutated to inject a sidecar anymore? If so, it'll break a lot of setups, e.g. for external-dns, which is not even close to getting support for Workload Identity, [PR](https://github.com/kubernetes-sigs/external-dns/pull/3111) has stayed unreviewed since October. Migration from adal to Azure SDK is not moving anywhere [either](https://github.com/kubernetes-sigs/external-dns/pull/3040).

**Comment by weisdd** [comment]
Had a call with AKS PM today. And the great news is that sidecar proxy injection is going to stay for now, so third-party projects still have some time left to embrace native support for Workload Identity.

**Comment by pinkfloydx33** [comment]
@asubmani Yes, it will work fine. We were part of private preview and that's exactly how we slowly migrated over. Nothing should change as they are two very different workflows that can coexist
